clang/lib/Sema/AnalysisBasedWarnings.cpp

```diff
@@ -399,6 +401,145 @@ static bool isNoexcept(const FunctionDecl *FD) {
   return false;
 }
 
+/// Checks if the given variable, which is assumed to be a function pointer, is
+/// initialized with a function having 'noreturn' attribute.
+static bool isInitializedWithNoReturn(const VarDecl *VD) {
```

Comment:
Hmm... We are checking the initializer here, but we don't check to make sure it wasn't assigned since then, right?  

IMO it seems this is a failure of `[[noreturn]]` not being part of the type :)